### PR TITLE
User profile with editable fields

### PR DESCRIPTION
**Pull request by SAISURYAtalla**
# Pull Request: User Profile Integration and Editable Features

## Description
This pull request enhances the Smart Insti App by introducing a user profile feature where a user can display his personal information along with his skills , Achievements . The user profile includes editable fields for description , skills section and achievements.

## Features
Displays all the information namely name , id, email , branch , graduation year , roles , about , skills as pie charts , and achievements as cards . 

## How to Test
1. Launch the Smart Insti App.
2. Navigate to the user profile screen by tapping on the user icon.
3. Explore the various sections of the user profile , along with the edit screen by tapping the edit icon on the top right.

## Screenshots 
user profile
edit_profile
home

Requesting review!
@henilp105

### CODE REVIEW

**Review by henilp105 — CHANGES_REQUESTED** [review]
Thanks @SAISURYAtalla, please clean up  the redundant code there is alot of redundant code.

File: frontend/lib/models/student2.dart [hunk 1]
```diff
@@ -0,0 +1,47 @@
+class Student {
```
[the remark this answers is not in the document]
Comment (SAISURYAtalla) [reply frontend/lib/models/student2.dart]: Sorry for the oversight , it is adjusted in the latest commit .

**Comment by SAISURYAtalla** [comment]
The changes requested have been made in the latest commit . 

Kindly review @henilp105

File: frontend/lib/constants/dummy_entries.dart [hunk 2]
```diff
@@ -251,6 +284,86 @@ class DummyStudents {
   ];
 }
 
+// class DummySkills {
```
Comment (henilp105) [inline frontend/lib/constants/dummy_entries.dart:287]: Remove redundant code.

File: frontend/lib/routes/routes.dart [hunk 3]
```diff
@@ -10,19 +10,27 @@ import '../screens/admin/add_menu.dart';
...
-      pageBuilder: (context, state) => MaterialPage(child: UserLogin()),
+      pageBuilder: (context, state) => const MaterialPage(child: Home()),
       routes: [
         GoRoute(
-          path: 'admin_login',
+          path: 'user_profile',
           pageBuilder: (context, state) =>
-              const MaterialPage(child: AdminLogin()),
+              const MaterialPage(child: UserProfile()),
+//       pageBuilder: (context, state) => MaterialPage(child: UserLogin()),
```
Comment (henilp105) [inline frontend/lib/routes/routes.dart:28]: Remove redundant code.

File: frontend/lib/routes/routes.dart [hunk 4]
```diff
@@ -80,3 +88,50 @@ final GoRouter routes = GoRouter(
     ),
   ],
 );
+
+// final GoRouter routes = GoRouter(
```
Comment (henilp105) [inline frontend/lib/routes/routes.dart:92]: Remove redundant code.

**Review by henilp105 — COMMENTED** [review]
thanks @SAISURYAtalla , Remove redundant code.

**Comment by henilp105** [comment]
@SAISURYAtalla resolve conflicts.

**Comment by SAISURYAtalla** [comment]
> @SAISURYAtalla resolve conflicts.

On it .

**Comment by SAISURYAtalla** [comment]
all conflicts resolved and profile updated along with edit profile and also integrated with auth and remaining PR's

as you can see , the details of the logged in user is only being displayed(which is mine )
Screenshot 2024-02-14 162203
Screenshot 2024-02-14 162216
Screenshot 2024-02-14 162225

Requesting re-review @henilp105 @Aditya062003

**Pull request merged**